tools/distro-json-generate.mjs

```diff
@@ -1,73 +1,106 @@
-  shell.echo('This script requires distro-info, exiting...');
-  shell.exit(2);
-}
+/**
+ * Converts valid CSV string into JSON.
+ * @param {string} raw CSV string
+ * @returns {string} JSON representation of input CSV
+ */
+function csvToJson(raw) {
+  const lines = raw.split(/\r?\n/);
+  const res = {};
+  const headers = lines[0].split(',');
 
-if (!shell.which(`sed`)) {
-  shell.echo('This script requires sed, exiting...');
-  shell.exit(2);
-}
+  // drop headers
+  lines.shift();
 
-shell.echo(`OK`);
+  // drop "version" header
+  headers.shift();
 
-const ubuntuDistroInfo = shell.exec(
-  `ubuntu-distro-info --all -f | sed -r 's/Ubuntu|"|LTS |Debian //g; s/([0-9]+.[0-9]+) /\\1 /; s/.*/\\L&/; s/( [a-z]*) [a-z]*/\\1/g; s/^[ \\t]*//'`,
-  { silent: true }
-);
+  for (const l of lines) {
+    if (!l) {
+      continue;
+    }
 
-/**
- * @param {string} str
- * @returns {{}}
- */
-function objectify(str) {
-  let obj = {};
-
-  for (const line of str.split(/\r?\n/)) {
-    let [ver, codename] = line.split(' ');
-    // eslint-disable-next-line
-    // @ts-ignore
-    obj[ver] = codename;
-  }
+    const obj = {};
+    const line = l.split(',');
+    let ver = line?.shift()?.replace(/LTS|\s/g, '');
 
-  return obj;
+    for (const [i, h] of headers.entries()) {
+      // eslint-disable-next-line
+      // @ts-ignore
+      obj[h.replace('-', '_')] = line[i];
+    }
+
+    if (ver) {
+      // Debian related, example codename "hamm" version is 2.0
+      // change 2.0 -> 2
+      ver = ver.endsWith('.0') ? ver.replace('.0', '') : ver;
+      // eslint-disable-next-line
+      // @ts-ignore
+      res[`v${ver}`] = obj;
+    }
+  }
+  return JSON.stringify(res, undefined, 2);
 }
 
 /**
- * @param {string} file
- * @param {string} newData
+ * Update given file with new provided data.
+ * @param {string} file Path to a data file
+ * @param {string} newData New data to be written
  */
 async function updateJsonFile(file, newData) {
   let oldData;
+  let newAdj;
 
   try {
     oldData = fs.existsSync(file) ? await fs.readFile(file, 'utf8') : null;
-    // Eliminate formatting
-    oldData = oldData?.replace(/\s/g, '') ?? null;
+    // Eliminate formatting. removes WS in the beginning, end. before & after non characters
+    oldData = oldData?.replace(/^\s|\s$|\B\s|\s\B/g, '') ?? null;
+    newAdj = newData?.replace(/^\s|\s$|\B\s|\s\B/g, '') ?? null;
   } catch (e) {
     shell.echo(e.toString());
     shell.exit(1);
   }
 
-  const parsedData = JSON.stringify(objectify(newData), undefined, 2);
-
-  if (oldData === parsedData) {
+  if (oldData === newAdj) {
     shell.echo(`${file} is up to date.`);
     return;
   }
 
+  const oldLen = oldData?.length ?? -1;
+  const newLen = newAdj?.length ?? -1;
+
+  if (oldLen === -1 || newLen === -1 || oldLen > newLen) {
+    shell.echo(`New data might be corrupted!`);
+    shell.echo(`Aborting ${file} update`);
+    shell.echo(`**************** NEW DATA ****************\n${newData} `);
+    return;
+  }
```

Comment:
I think we don't need this here, should be covered by normal unit tests.

Reply:
cool, thats right. never thought of it this way, 
removed.